numpy/lib/function_base.py

```diff
@@ -514,7 +514,9 @@ def average(a, axis=None, weights=None, returned=False):
         avg = a.mean(axis)
         scl = avg.dtype.type(a.size/avg.size)
     else:
-        a = a + 0.0
+        if np.issubclass_(a.dtype.type, np.inexact) is False:
```

Comment:
They are not the same.  In Python `is` checks to see if the two names point to the same object ("object identity").  So although using `is` works fine here, it wouldn't in other cases because it requires the LHS to be exactly the object `False`.  Using `not X` will coerce `X` to a bool, which is in general what you want.  

```
a = False

a is False
Out[2]: True

b = 0 

b is False
Out[4]: False

not b
Out[5]: True
```
I would also argue that `is False` or `is True` isn't really idiomatic.